sale_order_split_strategy/models/sale_order.py

```diff
@@ -0,0 +1,147 @@
+            section_lines = self.env["sale.order.line"].browse(line_ids)
+            for line in section_lines:
+                if line in lines_to_split:
+                    line.write(target_order._prepare_line_move_defaults())
+                # FIXME: We should consider if the line is part of the section
+                if copy_notes and line.display_type == "line_note":
+                    line.copy(target_order._prepare_line_move_defaults())
+            lines_in_section_to_split = [
+                li in lines_to_split for li in section_lines if not li.display_type
+            ]
+            # If all lines in section are to split, move the section to split order
+            if copy_sections and section_line and all(lines_in_section_to_split):
+                section_line.write(target_order._prepare_line_move_defaults())
+            # If any line in section is to split, copy the section to split order
+            elif copy_sections and section_line and any(lines_in_section_to_split):
+                section_line.copy(target_order._prepare_line_move_defaults())
+        return True
+
+    def _get_lines_grouped_by_sections(self):
+        # Prepare a dict from sale order lines and their sections
+        self.ensure_one()
+        sections_dict = defaultdict(list)
+        section_id = None
+        for line in self.order_line.sorted():
+            if line.display_type == "line_section":
+                section_id = line.id
+            sections_dict[section_id].append(line.id)
+        return sections_dict
+
+    def _prepare_message_split_from(self, order):
+        self.ensure_one()
+        template = self.env["ir.ui.view"].search(
+            [("key", "=", "sale_order_split_strategy.split_from")], limit=1
+        )
+        if not template:
+            return _(
+                "This sale order was created after splitting lines from %(order_name)s"
+                " using strategy %(strategy)s",
+                order_name=order.name,
+                strategy=order.split_strategy_id.name,
+            )
+        else:
+            return template._render(values={"from_order": order})
+
+    def _prepare_message_split_to(self, order):
+        self.ensure_one()
+        template = self.env["ir.ui.view"].search(
+            [("key", "=", "sale_order_split_strategy.split_to")], limit=1
+        )
+        if not template:
+            return _(
+                "This sale order had some of its lines split to %(order_name)s"
+                " using strategy %(strategy)s",
+                order_name=order.name,
+                strategy=self.split_strategy_id.name,
+            )
+        else:
+            return template._render(values={"from_order": self, "to_order": order})
+
+    def _prepare_order_copy_defaults(self):
+        """Hook to customize values used on new sale order from split"""
+        self.ensure_one()
+        return {"order_line": False}
+
+    def _prepare_line_move_defaults(self):
```

Comment:
No risk to introduce conflicts with method names, depending on the set of modules installed? I would have named it `_prepare_line_split_defaults` perhaps